app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1547,33 +1567,37 @@ class BrowserTabViewModel(
         }
     }
 
-    override fun onSavedSiteEdited(savedSite: SavedSite) {
-        when (savedSite) {
-            is SavedSite.Bookmark -> {
-                viewModelScope.launch(dispatchers.io()) {
-                    editBookmark(savedSite)
-                }
-            }
-            is SavedSite.Favorite -> {
-                viewModelScope.launch(dispatchers.io()) {
-                    editFavorite(savedSite)
-                }
-            }
+    override fun onSavedSiteFavoriteEdited(favorite: SavedSite.Favorite) {
+        viewModelScope.launch(dispatchers.io()) {
+            editFavorite(favorite)
         }
     }
 
-    fun onEditSavedSiteRequested(savedSite: SavedSite) {
-        command.value = ShowEditSavedSiteDialog(savedSite)
+    override fun onSavedSiteBookmarkEdited(bookmark: SavedSite.Bookmark, bookmarkFolder: BookmarkFolder) {
```

Comment:
I don't think we need bookmarkFolder as parameter here. I would prefer to keep one single method and get benefit of `SavedSite` sealed class.

What about just receiving `bookmark`, and retrieve `bookmarkFolder` from db inside `editBookmark`?

Reply:
Sure, this can be done. 

Trying this right away.